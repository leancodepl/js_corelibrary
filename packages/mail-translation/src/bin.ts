#!/usr/bin/env node
/* eslint-disable no-console */

import { Command } from 'commander';
import * as path from 'path';
import { CliConfig, loadConfig, loadConfigFromFile, mergeWithDefaults, OutputMode, validateConfig } from './lib/configLoader';
import { MailTranslator } from './lib/mailTranslator';

const program = new Command();

interface CliOptions {
  config?: string;
  translationsPath?: string;
  mailsPath?: string;
  outputPath?: string;
  outputMode?: OutputMode;
  defaultLanguage?: string;
  language?: string;
  languages?: string;
  verbose?: boolean;
  beautify?: boolean;
  minify?: boolean;
  validationLevel?: 'skip' | 'soft' | 'strict';
}

async function translateMails(options: CliOptions): Promise<void> {
  try {
    let config: CliConfig;

    // Load configuration
    if (options.config) {
      config = await loadConfigFromFile(options.config);
    } else {
      const loadedConfig = await loadConfig();
      config = loadedConfig || {} as CliConfig;
    }

    // Override config with CLI options
    if (options.translationsPath) config.translationsPath = options.translationsPath;
    if (options.mailsPath) config.mailsPath = options.mailsPath;
    if (options.outputPath) config.outputPath = options.outputPath;
    if (options.outputMode) config.outputMode = options.outputMode;
    if (options.defaultLanguage) config.defaultLanguage = options.defaultLanguage;
    if (options.verbose !== undefined) config.verbose = options.verbose;

    // Handle MJML options
    if (options.beautify !== undefined) {
      config.mjmlOptions = config.mjmlOptions || {};
      config.mjmlOptions.beautify = options.beautify;
    }
    if (options.minify !== undefined) {
      config.mjmlOptions = config.mjmlOptions || {};
      config.mjmlOptions.minify = options.minify;
    }
    if (options.validationLevel) {
      config.mjmlOptions = config.mjmlOptions || {};
      config.mjmlOptions.validationLevel = options.validationLevel;
    }

    // Merge with defaults
    config = mergeWithDefaults(config);

    // Validate configuration
    validateConfig(config);

    if (config.verbose) {
      console.log('Configuration:', JSON.stringify(config, null, 2));
    }

    // Initialize translator
    const translator = new MailTranslator({
      translationsPath: config.translationsPath,
      mailsPath: config.mailsPath,
      outputPath: config.outputPath,
      outputMode: config.outputMode,
      defaultLanguage: config.defaultLanguage,
      mjmlOptions: config.mjmlOptions,
    });

    await translator.initialize();

    if (config.verbose) {
      console.log('Available languages:', translator.getAvailableLanguages());
    }

    // Check if we have any translations loaded
    if (!translator.hasTranslations()) {
      console.log('No translations found. Processing templates without translations.');
    }

    // Determine which languages to process
    let languagesToProcess: string[];
    if (options.language) {
      languagesToProcess = [options.language];
    } else if (options.languages) {
      languagesToProcess = options.languages.split(',').map(lang => lang.trim());
    } else if (config.languages) {
      languagesToProcess = config.languages;
    } else {
      languagesToProcess = translator.getAvailableLanguages();
    }

    // Validate languages
    const availableLanguages = translator.getAvailableLanguages();
    for (const lang of languagesToProcess) {
      if (!availableLanguages.includes(lang)) {
        throw new Error(`Language '${lang}' not found. Available languages: ${availableLanguages.join(', ')}`);
      }
    }

    console.log(`Processing languages: ${languagesToProcess.join(', ')}`);
    console.log(`Output mode: ${config.outputMode}`);
    console.log(`Default language: ${config.defaultLanguage}`);

    // Process each language
    const allTranslatedMails: { [language: string]: any[] } = {};
    
    for (const language of languagesToProcess) {
      if (config.verbose) {
        console.log(`Processing language: ${language}`);
      }
      
      const translatedMails = await translator.translateAllTemplates(language);
      allTranslatedMails[language] = translatedMails;
      
      console.log(`✓ ${language}: ${translatedMails.length} templates processed`);
      
      // Log errors if any
      for (const mail of translatedMails) {
        if (mail.errors.length > 0) {
          console.warn(`  ⚠ ${mail.name}: ${mail.errors.length} errors`);
          if (config.verbose) {
            mail.errors.forEach((error: any) => {
              console.warn(`    - Line ${error.line}: ${error.message}`);
            });
          }
        }
      }
    }

    // Save translated mails
    if (config.outputPath) {
      await translator.saveTranslatedMails(allTranslatedMails);
      console.log(`✓ Translated mails saved to: ${path.resolve(config.outputPath)}`);
    }

    console.log('Mail translation completed successfully!');

  } catch (error) {
    console.error('Error:', error instanceof Error ? error.message : error);
    process.exit(1);
  }
}

// CLI setup
program
  .name('@leancodepl/mail-translation')
  .description('CLI tool for translating MJML email templates')
  .version('0.0.1');

program
  .command('translate')
  .description('Translate mail templates')
  .option('-c, --config <path>', 'Path to configuration file')
  .option('-t, --translations-path <path>', 'Path to translations directory')
  .option('-m, --mails-path <path>', 'Path to mail templates directory')
  .option('-o, --output-path <path>', 'Path to output directory')
  .option('--output-mode <mode>', 'Output mode (kratos|razor)', 'kratos')
  .option('--default-language <lang>', 'Default language for templates', 'en')
  .option('-l, --language <lang>', 'Process only specific language')
  .option('--languages <langs>', 'Process specific languages (comma-separated)')
  .option('-v, --verbose', 'Enable verbose output')
  .option('--beautify', 'Beautify HTML output')
  .option('--minify', 'Minify HTML output')
  .option('--validation-level <level>', 'MJML validation level (strict|soft|skip)', 'soft')
  .action(translateMails);

// Set default command to translate
program.arguments('[command]').action((cmd) => {
  if (cmd) {
    program.outputHelp();
    process.exit(1);
  } else {
    // If no command is provided, run translate with no options
    translateMails({}).catch((error) => {
      console.error('Error:', error instanceof Error ? error.message : error);
      process.exit(1);
    });
  }
});

// Parse arguments
program.parse();

// If no arguments provided, show help
if (process.argv.length === 2) {
  program.outputHelp();
} 