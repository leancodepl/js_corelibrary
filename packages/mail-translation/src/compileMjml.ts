const mjml = require("mjml")

export interface MjmlCompileOptions {
    keepComments?: boolean
    beautify?: boolean
    minify?: boolean
    validationLevel?: "skip" | "soft" | "strict"
    filePath?: string
}

export interface MjmlCompileResult {
    html: string
    errors: Array<{
        line: number
        message: string
        tagName: string
    }>
}

export function compileMjml(mjmlContent: string, options: MjmlCompileOptions = {}): MjmlCompileResult {
    const defaultOptions: MjmlCompileOptions = {
        keepComments: false,
        beautify: false,
        minify: false,
        validationLevel: "soft",
        filePath: ".",
        ...options,
    }

    try {
        const result = mjml(mjmlContent, defaultOptions)

        return {
            html: result.html,
            errors: result.errors || [],
        }
    } catch (error) {
        throw new Error(`MJML compilation failed: ${error}`)
    }
}
