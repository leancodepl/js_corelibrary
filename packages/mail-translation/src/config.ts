import { z } from "zod/v4"

const outputModeSchema = z.enum(["kratos", "razor"])

export type OutputMode = z.infer<typeof outputModeSchema>

export const mailTranslationConfigSchema = z.object({
  translationsPath: z
    .string()
    .optional()
    .describe(
      "Path to directory containing translation JSON files. When omitted, templates are compiled without translations. Each JSON file should be named with the language code (e.g., en.json, pl.json).",
    ),
  mailsPath: z
    .string()
    .describe(
      "Path to directory containing MJML email templates. All .mjml files in this directory will be processed.",
    ),
  plaintextMailsPath: z
    .string()
    .optional()
    .describe(
      "Path to directory containing plaintext templates. If not specified, defaults to the same value as mailsPath. Used for generating text-only versions of emails alongside HTML versions.",
    ),
  outputPath: z
    .string()
    .describe(
      "Directory where processed templates will be saved. The tool will create this directory if it doesn't exist.",
    ),
  outputMode: outputModeSchema.describe(
    "Target templating system format: 'kratos' for Go template files compatible with Ory Kratos, 'razor' for C# Razor template files.",
  ),
  defaultLanguage: z
    .string()
    .optional()
    .describe("Default language code for templates with translations. Required when translationsPath is provided."),
  languages: z
    .array(z.string())
    .optional()
    .describe(
      "Array of language codes to process. When omitted, all languages found in translation files are automatically processed. Use this to limit output to specific languages.",
    ),
  kratosLanguageVariable: z
    .string()
    .optional()
    .describe(
      "Variable path used for language detection in Kratos templates. Defaults to '.Identity.traits.lang'. This determines how the generated template will access the user's language preference. Only used in Kratos mode.",
    ),
})

export type MailTranslationConfig = z.infer<typeof mailTranslationConfigSchema>
