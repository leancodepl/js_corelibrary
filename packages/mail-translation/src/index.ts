export * from "./lib/mail-translation";
