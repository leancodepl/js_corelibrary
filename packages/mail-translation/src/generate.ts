import { MailTranslationConfig } from "./config"
import { loadMjmlTemplates, loadPlaintextTemplates } from "./loadTemplates"
import { loadTranslations } from "./loadTranslations"
import { processTemplate } from "./processTemplate"

export async function generate(config: Omit<MailTranslationConfig, "outputPath">) {
  const translationData = await loadTranslations(config.translationsPath)

  const mjmlTemplates = await loadMjmlTemplates(config.mailsPath)
  const plaintextTemplates = config.plaintextMailsPath
    ? await loadPlaintextTemplates({
        plaintextMailsPath: config.plaintextMailsPath,
        outputMode: config.outputMode,
      })
    : []

  return [...mjmlTemplates, ...plaintextTemplates].map(template =>
    processTemplate({
      template,
      translationData,
      outputMode: config.outputMode,
      defaultLanguage: config.defaultLanguage,
      kratosLanguageVariable: config.kratosLanguageVariable,
      mailsPath: config.mailsPath,
    }),
  )
}
