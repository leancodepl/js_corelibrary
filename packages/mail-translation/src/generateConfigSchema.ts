import { writeFileSync } from "fs"
import { join } from "path"
import { z } from "zod/v4"
import { mailTranslationConfigSchema } from "./config"

const jsonSchema = z.toJSONSchema(mailTranslationConfigSchema)

const output = {
  $schema: "http://json-schema.org/draft-07/schema#",
  title: "@leancodepl/mail-translation Configuration",
  description: "Schema for @leancodepl/mail-translation configuration",
  ...jsonSchema,
}

const packageRoot = join(__dirname, "..")
const outputPath = join(packageRoot, "schema.json")

writeFileSync(outputPath, JSON.stringify(output, null, 2) + '\n')
