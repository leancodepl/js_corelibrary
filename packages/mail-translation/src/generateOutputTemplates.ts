import { OutputMode } from "./config"
import { generateKratosOutputTemplates } from "./generateKratosOutputTemplates"
import { generateRazorOutputTemplates } from "./generateRazorOutputTemplates"
import { TranslatedTemplate } from "./processTemplate"

export interface OutputTemplate {
    filename: string
    content: string
}

export function generateOutputTemplates(
    translatedTemplates: TranslatedTemplate[],
    outputMode: OutputMode,
    defaultLanguage: string,
): OutputTemplate[] {
    switch (outputMode) {
        case "kratos":
            return generateKratosOutputTemplates({ translatedTemplates, defaultLanguage })
        case "razor":
            return generateRazorOutputTemplates({ translatedTemplates, defaultLanguage })
    }
}
