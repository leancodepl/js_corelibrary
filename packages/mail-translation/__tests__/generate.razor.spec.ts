import * as path from "path"
import { generate } from "../src"

const razorTestDir = path.join(__dirname, "razor")

describe("generate function - Razor mode snapshots", () => {
    const config = {
        translationsPath: path.join(razorTestDir, "translations"),
        mailsPath: path.join(razorTestDir, "mails"),
        plaintextMailsPath: path.join(razorTestDir, "mails"),
        outputMode: "razor" as const,
        defaultLanguage: "en",
    }

    it("should generate razor templates matching snapshots", async () => {
        const result = await generate(config)
        
        // Sort results for consistent snapshots
        const sortedResult = result
            .map(template => ({
                ...template,
                outputTemplates: template.outputTemplates.sort((a, b) => a.filename.localeCompare(b.filename))
            }))
            .sort((a, b) => a.name.localeCompare(b.name))

        expect(sortedResult).toMatchSnapshot()
    })

    it("should preserve razor syntax in templates", async () => {
        const result = await generate(config)
        
        const notificationTemplate = result.find(template => template.name === "notification")
        const htmlTemplate = notificationTemplate?.outputTemplates.find(t => t.filename === "notification.cshtml")
        
        expect(htmlTemplate?.content).toContain("@Model.User.FullName")
        expect(htmlTemplate?.content).toContain("@Model.Status")
        expect(htmlTemplate?.content).toContain("@Model.ActionUrl")
        expect(htmlTemplate?.content).toContain("@Model.ReferenceNumber")
    })
}) 