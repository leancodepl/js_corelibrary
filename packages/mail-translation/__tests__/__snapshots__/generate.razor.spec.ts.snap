// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`generate function - Razor mode should generate razor templates matching snapshots 1`] = `
[
  {
    "mjmlParseErrors": [],
    "name": "notification",
    "outputTemplates": [
      {
        "content": "<!doctype html>
<html lang="und" dir="auto" xmlns="http://www.w3.org/1999/xhtml" xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office">

<head>
  <title></title>
  <!--[if !mso]><!-->
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <!--<![endif]-->
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <style type="text/css">
    #outlook a {
      padding: 0;
    }

    body {
      margin: 0;
      padding: 0;
      -webkit-text-size-adjust: 100%;
      -ms-text-size-adjust: 100%;
    }

    table,
    td {
      border-collapse: collapse;
      mso-table-lspace: 0pt;
      mso-table-rspace: 0pt;
    }

    img {
      border: 0;
      height: auto;
      line-height: 100%;
      outline: none;
      text-decoration: none;
      -ms-interpolation-mode: bicubic;
    }

    p {
      display: block;
      margin: 13px 0;
    }
  </style>
  <!--[if mso]>
    <noscript>
    <xml>
    <o:OfficeDocumentSettings>
      <o:AllowPNG/>
      <o:PixelsPerInch>96</o:PixelsPerInch>
    </o:OfficeDocumentSettings>
    </xml>
    </noscript>
    <![endif]-->
  <!--[if lte mso 11]>
    <style type="text/css">
      .mj-outlook-group-fix { width:100% !important; }
    </style>
    <![endif]-->

  <!--[if !mso]><!-->
  <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet" type="text/css">
  <style type="text/css">
    @@import url(https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap);
  </style>
  <!--<![endif]-->

  <style type="text/css">
    @@media only screen and (min-width:480px) {
      .mj-column-per-100 {
        width: 100% !important;
        max-width: 100%;
      }
    }
  </style>
  <style media="screen and (min-width:480px)">
    .moz-text-html .mj-column-per-100 {
      width: 100% !important;
      max-width: 100%;
    }
  </style>

  <style type="text/css">
    @@media only screen and (max-width:479px) {
      table.mj-full-width-mobile {
        width: 100% !important;
      }

      td.mj-full-width-mobile {
        width: auto !important;
      }
    }
  </style>

</head>

<body style="word-spacing:normal;background-color:#f9fafb;">

  <div style="background-color:#f9fafb;" lang="und" dir="auto">

    <!--[if mso | IE]><table align="center" border="0" cellpadding="0" cellspacing="0" class="header-outlook" role="presentation" style="width:600px;" width="600" ><tr><td style="line-height:0px;font-size:0px;mso-line-height-rule:exactly;"><![endif]-->

    <div class="header" style="background-color: #059669; color: white; margin: 0px auto; max-width: 600px;">

      <table align="center" border="0" cellpadding="0" cellspacing="0" role="presentation" style="width:100%;">
        <tbody>
          <tr>
            <td style="direction:ltr;font-size:0px;padding:0;text-align:center;">
              <!--[if mso | IE]><table role="presentation" border="0" cellpadding="0" cellspacing="0"><tr><td class="" style="vertical-align:top;width:600px;" ><![endif]-->

              <div class="mj-column-per-100 mj-outlook-group-fix" style="font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;">

                <table border="0" cellpadding="0" cellspacing="0" role="presentation" width="100%">
                  <tbody>
                    <tr>
                      <td style="vertical-align:top;padding:0;">

                        <table border="0" cellpadding="0" cellspacing="0" role="presentation" style width="100%">
                          <tbody>

                            <tr>
                              <td align="center" style="font-size:0px;padding:0;padding-bottom:24px;word-break:break-word;">

                                <table border="0" cellpadding="0" cellspacing="0" role="presentation" style="border-collapse:collapse;border-spacing:0px;">
                                  <tbody>
                                    <tr>
                                      <td style="width:150px;">

                                        <img alt="System Notification" src="\${storage}/static/company-logo.png" style="border:0;display:block;outline:none;text-decoration:none;height:auto;width:100%;font-size:13px;" width="150" height="auto">

                                      </td>
                                    </tr>
                                  </tbody>
                                </table>

                              </td>
                            </tr>

                            <tr>
                              <td align="left" style="font-size:0px;padding:0;padding-bottom:16px;word-break:break-word;">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:20px;font-weight:700;line-height:28px;text-align:left;color:white;">System Notification</div>

                              </td>
                            </tr>

                          </tbody>
                        </table>

                      </td>
                    </tr>
                  </tbody>
                </table>

              </div>

              <!--[if mso | IE]></td></tr></table><![endif]-->
            </td>
          </tr>
        </tbody>
      </table>

    </div>

    <!--[if mso | IE]></td></tr></table><table align="center" border="0" cellpadding="0" cellspacing="0" class="" role="presentation" style="width:600px;" width="600" ><tr><td style="line-height:0px;font-size:0px;mso-line-height-rule:exactly;"><![endif]-->

    <div style="margin:0px auto;max-width:600px;">

      <table align="center" border="0" cellpadding="0" cellspacing="0" role="presentation" style="width:100%;">
        <tbody>
          <tr>
            <td style="direction:ltr;font-size:0px;padding:20px;text-align:center;">
              <!--[if mso | IE]><table role="presentation" border="0" cellpadding="0" cellspacing="0"><tr><td class="" style="vertical-align:top;width:560px;" ><![endif]-->

              <div class="mj-column-per-100 mj-outlook-group-fix" style="font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;">

                <table border="0" cellpadding="0" cellspacing="0" role="presentation" width="100%">
                  <tbody>
                    <tr>
                      <td style="vertical-align:top;padding:0;">

                        <table border="0" cellpadding="0" cellspacing="0" role="presentation" style width="100%">
                          <tbody>

                            <tr>
                              <td align="left" style="font-size:0px;padding:0;padding-bottom:20px;word-break:break-word;">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:14px;font-weight:400;line-height:20px;text-align:left;color:#374151;">Dear @Model.User.FullName,</div>

                              </td>
                            </tr>

                            <tr>
                              <td align="left" class="alert-box" style="background-color: #fef3c7; border-left: 4px solid #f59e0b; font-size: 0px; padding: 0; padding-bottom: 20px; word-break: break-word;" bgcolor="#fef3c7">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:14px;font-weight:400;line-height:20px;text-align:left;color:#374151;">Your account status has been updated to: @Model.Status as of @Model.UpdateDate.</div>

                              </td>
                            </tr>

                            <tr>
                              <td align="left" style="font-size:0px;padding:0;padding-bottom:20px;word-break:break-word;">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:14px;font-weight:400;line-height:20px;text-align:left;color:#374151;">Action required: @Model.RequiredAction</div>

                              </td>
                            </tr>

                            <tr>
                              <td align="center" style="font-size:0px;padding:0;word-break:break-word;">

                                <table border="0" cellpadding="0" cellspacing="0" role="presentation" style="border-collapse:separate;line-height:100%;">
                                  <tbody>
                                    <tr>
                                      <td align="center" bgcolor="#2563eb" role="presentation" style="border:none;border-radius:6px;cursor:auto;mso-padding-alt:10px 16px;background:#2563eb;" valign="middle">
                                        <a href="@Model.ActionUrl" style="display: inline-block; background: #2563eb; color: #ffffff; font-family: Roboto, Arial, sans-serif; font-size: 14px; font-weight: 500; line-height: 20px; margin: 0; text-decoration: none; text-transform: none; padding: 10px 16px; mso-padding-alt: 0px; border-radius: 6px;" target="_blank">
                                          Take Action Now
                                        </a>
                                      </td>
                                    </tr>
                                  </tbody>
                                </table>

                              </td>
                            </tr>

                            <tr>
                              <td align="center" style="font-size:0px;padding:0;padding-bottom:20px;word-break:break-word;">

                                <p style="border-top:solid 1px #d1d5db;font-size:1px;margin:0px auto;width:100%;">
                                </p>

                                <!--[if mso | IE]><table align="center" border="0" cellpadding="0" cellspacing="0" style="border-top:solid 1px #d1d5db;font-size:1px;margin:0px auto;width:560px;" role="presentation" width="560px" ><tr><td style="height:0;line-height:0;"> &nbsp;
</td></tr></table><![endif]-->

                              </td>
                            </tr>

                            <tr>
                              <td align="center" style="font-size:0px;padding:0;padding-bottom:16px;word-break:break-word;">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:12px;font-weight:400;line-height:18px;text-align:center;color:#6b7280;">Reference: @Model.ReferenceNumber</div>

                              </td>
                            </tr>

                            <tr>
                              <td align="center" style="font-size:0px;padding:0;word-break:break-word;">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:11px;font-weight:400;line-height:16px;text-align:center;color:#9ca3af;">Need help? Contact our support team at support@example.com</div>

                              </td>
                            </tr>

                          </tbody>
                        </table>

                      </td>
                    </tr>
                  </tbody>
                </table>

              </div>

              <!--[if mso | IE]></td></tr></table><![endif]-->
            </td>
          </tr>
        </tbody>
      </table>

    </div>

    <!--[if mso | IE]></td></tr></table><![endif]-->

  </div>

</body>

</html>",
        "filename": "notification.cshtml",
      },
      {
        "content": "<!doctype html>
<html lang="und" dir="auto" xmlns="http://www.w3.org/1999/xhtml" xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office">

<head>
  <title></title>
  <!--[if !mso]><!-->
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <!--<![endif]-->
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <style type="text/css">
    #outlook a {
      padding: 0;
    }

    body {
      margin: 0;
      padding: 0;
      -webkit-text-size-adjust: 100%;
      -ms-text-size-adjust: 100%;
    }

    table,
    td {
      border-collapse: collapse;
      mso-table-lspace: 0pt;
      mso-table-rspace: 0pt;
    }

    img {
      border: 0;
      height: auto;
      line-height: 100%;
      outline: none;
      text-decoration: none;
      -ms-interpolation-mode: bicubic;
    }

    p {
      display: block;
      margin: 13px 0;
    }
  </style>
  <!--[if mso]>
    <noscript>
    <xml>
    <o:OfficeDocumentSettings>
      <o:AllowPNG/>
      <o:PixelsPerInch>96</o:PixelsPerInch>
    </o:OfficeDocumentSettings>
    </xml>
    </noscript>
    <![endif]-->
  <!--[if lte mso 11]>
    <style type="text/css">
      .mj-outlook-group-fix { width:100% !important; }
    </style>
    <![endif]-->

  <!--[if !mso]><!-->
  <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet" type="text/css">
  <style type="text/css">
    @@import url(https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap);
  </style>
  <!--<![endif]-->

  <style type="text/css">
    @@media only screen and (min-width:480px) {
      .mj-column-per-100 {
        width: 100% !important;
        max-width: 100%;
      }
    }
  </style>
  <style media="screen and (min-width:480px)">
    .moz-text-html .mj-column-per-100 {
      width: 100% !important;
      max-width: 100%;
    }
  </style>

  <style type="text/css">
    @@media only screen and (max-width:479px) {
      table.mj-full-width-mobile {
        width: 100% !important;
      }

      td.mj-full-width-mobile {
        width: auto !important;
      }
    }
  </style>

</head>

<body style="word-spacing:normal;background-color:#f9fafb;">

  <div style="background-color:#f9fafb;" lang="und" dir="auto">

    <!--[if mso | IE]><table align="center" border="0" cellpadding="0" cellspacing="0" class="header-outlook" role="presentation" style="width:600px;" width="600" ><tr><td style="line-height:0px;font-size:0px;mso-line-height-rule:exactly;"><![endif]-->

    <div class="header" style="background-color: #059669; color: white; margin: 0px auto; max-width: 600px;">

      <table align="center" border="0" cellpadding="0" cellspacing="0" role="presentation" style="width:100%;">
        <tbody>
          <tr>
            <td style="direction:ltr;font-size:0px;padding:0;text-align:center;">
              <!--[if mso | IE]><table role="presentation" border="0" cellpadding="0" cellspacing="0"><tr><td class="" style="vertical-align:top;width:600px;" ><![endif]-->

              <div class="mj-column-per-100 mj-outlook-group-fix" style="font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;">

                <table border="0" cellpadding="0" cellspacing="0" role="presentation" width="100%">
                  <tbody>
                    <tr>
                      <td style="vertical-align:top;padding:0;">

                        <table border="0" cellpadding="0" cellspacing="0" role="presentation" style width="100%">
                          <tbody>

                            <tr>
                              <td align="center" style="font-size:0px;padding:0;padding-bottom:24px;word-break:break-word;">

                                <table border="0" cellpadding="0" cellspacing="0" role="presentation" style="border-collapse:collapse;border-spacing:0px;">
                                  <tbody>
                                    <tr>
                                      <td style="width:150px;">

                                        <img alt="Powiadomienie systemowe" src="\${storage}/static/company-logo.png" style="border:0;display:block;outline:none;text-decoration:none;height:auto;width:100%;font-size:13px;" width="150" height="auto">

                                      </td>
                                    </tr>
                                  </tbody>
                                </table>

                              </td>
                            </tr>

                            <tr>
                              <td align="left" style="font-size:0px;padding:0;padding-bottom:16px;word-break:break-word;">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:20px;font-weight:700;line-height:28px;text-align:left;color:white;">Powiadomienie systemowe</div>

                              </td>
                            </tr>

                          </tbody>
                        </table>

                      </td>
                    </tr>
                  </tbody>
                </table>

              </div>

              <!--[if mso | IE]></td></tr></table><![endif]-->
            </td>
          </tr>
        </tbody>
      </table>

    </div>

    <!--[if mso | IE]></td></tr></table><table align="center" border="0" cellpadding="0" cellspacing="0" class="" role="presentation" style="width:600px;" width="600" ><tr><td style="line-height:0px;font-size:0px;mso-line-height-rule:exactly;"><![endif]-->

    <div style="margin:0px auto;max-width:600px;">

      <table align="center" border="0" cellpadding="0" cellspacing="0" role="presentation" style="width:100%;">
        <tbody>
          <tr>
            <td style="direction:ltr;font-size:0px;padding:20px;text-align:center;">
              <!--[if mso | IE]><table role="presentation" border="0" cellpadding="0" cellspacing="0"><tr><td class="" style="vertical-align:top;width:560px;" ><![endif]-->

              <div class="mj-column-per-100 mj-outlook-group-fix" style="font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;">

                <table border="0" cellpadding="0" cellspacing="0" role="presentation" width="100%">
                  <tbody>
                    <tr>
                      <td style="vertical-align:top;padding:0;">

                        <table border="0" cellpadding="0" cellspacing="0" role="presentation" style width="100%">
                          <tbody>

                            <tr>
                              <td align="left" style="font-size:0px;padding:0;padding-bottom:20px;word-break:break-word;">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:14px;font-weight:400;line-height:20px;text-align:left;color:#374151;">Szanowny/a @Model.User.FullName,</div>

                              </td>
                            </tr>

                            <tr>
                              <td align="left" class="alert-box" style="background-color: #fef3c7; border-left: 4px solid #f59e0b; font-size: 0px; padding: 0; padding-bottom: 20px; word-break: break-word;" bgcolor="#fef3c7">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:14px;font-weight:400;line-height:20px;text-align:left;color:#374151;">Status Twojego konta został zaktualizowany na: @Model.Status z dniem @Model.UpdateDate.</div>

                              </td>
                            </tr>

                            <tr>
                              <td align="left" style="font-size:0px;padding:0;padding-bottom:20px;word-break:break-word;">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:14px;font-weight:400;line-height:20px;text-align:left;color:#374151;">Wymagane działanie: @Model.RequiredAction</div>

                              </td>
                            </tr>

                            <tr>
                              <td align="center" style="font-size:0px;padding:0;word-break:break-word;">

                                <table border="0" cellpadding="0" cellspacing="0" role="presentation" style="border-collapse:separate;line-height:100%;">
                                  <tbody>
                                    <tr>
                                      <td align="center" bgcolor="#2563eb" role="presentation" style="border:none;border-radius:6px;cursor:auto;mso-padding-alt:10px 16px;background:#2563eb;" valign="middle">
                                        <a href="@Model.ActionUrl" style="display: inline-block; background: #2563eb; color: #ffffff; font-family: Roboto, Arial, sans-serif; font-size: 14px; font-weight: 500; line-height: 20px; margin: 0; text-decoration: none; text-transform: none; padding: 10px 16px; mso-padding-alt: 0px; border-radius: 6px;" target="_blank">
                                          Wykonaj działanie teraz
                                        </a>
                                      </td>
                                    </tr>
                                  </tbody>
                                </table>

                              </td>
                            </tr>

                            <tr>
                              <td align="center" style="font-size:0px;padding:0;padding-bottom:20px;word-break:break-word;">

                                <p style="border-top:solid 1px #d1d5db;font-size:1px;margin:0px auto;width:100%;">
                                </p>

                                <!--[if mso | IE]><table align="center" border="0" cellpadding="0" cellspacing="0" style="border-top:solid 1px #d1d5db;font-size:1px;margin:0px auto;width:560px;" role="presentation" width="560px" ><tr><td style="height:0;line-height:0;"> &nbsp;
</td></tr></table><![endif]-->

                              </td>
                            </tr>

                            <tr>
                              <td align="center" style="font-size:0px;padding:0;padding-bottom:16px;word-break:break-word;">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:12px;font-weight:400;line-height:18px;text-align:center;color:#6b7280;">Numer referencyjny: @Model.ReferenceNumber</div>

                              </td>
                            </tr>

                            <tr>
                              <td align="center" style="font-size:0px;padding:0;word-break:break-word;">

                                <div style="font-family:Roboto, Arial, sans-serif;font-size:11px;font-weight:400;line-height:16px;text-align:center;color:#9ca3af;">Potrzebujesz pomocy? Skontaktuj się z naszym zespołem wsparcia: support@example.com</div>

                              </td>
                            </tr>

                          </tbody>
                        </table>

                      </td>
                    </tr>
                  </tbody>
                </table>

              </div>

              <!--[if mso | IE]></td></tr></table><![endif]-->
            </td>
          </tr>
        </tbody>
      </table>

    </div>

    <!--[if mso | IE]></td></tr></table><![endif]-->

  </div>

</body>

</html>",
        "filename": "notification.pl.cshtml",
      },
    ],
  },
  {
    "mjmlParseErrors": [],
    "name": "notification",
    "outputTemplates": [
      {
        "content": "Powiadomienie systemowe

Szanowny/a @Model.User.FullName,

Status Twojego konta został zaktualizowany na: @Model.Status z dniem @Model.UpdateDate.

Wymagane działanie: @Model.RequiredAction

Wykonaj działanie teraz: @Model.ActionUrl

Numer referencyjny: @Model.ReferenceNumber

Potrzebujesz pomocy? Skontaktuj się z naszym zespołem wsparcia: support@example.com",
        "filename": "notification.pl.txt.cshtml",
      },
      {
        "content": "System Notification

Dear @Model.User.FullName,

Your account status has been updated to: @Model.Status as of @Model.UpdateDate.

Action required: @Model.RequiredAction

Take Action Now: @Model.ActionUrl

Reference: @Model.ReferenceNumber

Need help? Contact our support team at support@example.com",
        "filename": "notification.txt.cshtml",
      },
    ],
  },
]
`;
